hack/vendor.sh

```diff
@@ -1,14 +1,41 @@
 #!/usr/bin/env bash
-
-# This file is just wrapper around 'go mod vendor' tool.
+#
+# This file is just a wrapper around the 'go mod vendor' tool.
 # For updating dependencies you should change `vendor.mod` file in root of the
 # project.
 
 set -e
-set -x
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 "${SCRIPTDIR}"/go-mod-prepare.sh
 
-GO111MODULE=auto go mod tidy -modfile 'vendor.mod' -compat 1.18
-GO111MODULE=auto go mod vendor -modfile vendor.mod
+export GO111MODULE=on
+
+tidy() {
+	(
+		set -x
+		go mod tidy -modfile vendor.mod -compat 1.18
+	)
+}
+
+vendor() {
+	(
+		set -x
+		go mod vendor -modfile vendor.mod
+	)
+}
+
+help() {
+	printf "%s:\n" "$(basename "$0")"
+	echo "  - tidy: run go mod tidy"
+	echo "  - vendor: run go mod vendor"
+	echo "  - all: run tidy && vendor"
+	echo "  - help: show this help"
+}
```

Comment:
We should really have a `vendor` (and `tidy` ?) target in the makefile, so that we have a canonical "help" (we already have  `make help`).

Reply:
The current docs say to use `hack/vendor.sh` -- I'd rather slowly chip away at technical debt in the build system and think about what would be more maintainable in the long term without adding too much that is "new" before we expose more/depend more on these implementation details.

That all goes to say that I think the status quo here is fine and the long term intention is to replace/refine this further, as we have logic spread out among Dockerfiles, shell scripts, Makefiles, and more, and it's very hard to sustain/reason about.